pypeit/bspline/utilc.py

```diff
@@ -164,17 +164,20 @@ def solution_arrays(nn, npoly, nord, ydata, action, ivar, upper, lower):
     """
     nfull = nn * npoly
     bw = npoly * nord
-    # NOTE: Declared as empty because the c code zeros them out
-    alpha = np.empty((bw, nfull+bw), dtype=float)
-    beta = np.empty((nfull+bw,), dtype=float)
+
+    alpha = np.zeros((bw, nfull+bw), dtype=float)
+    beta = np.zeros((nfull+bw,), dtype=float)
     upper = np.array(upper, dtype=np.int64)
     lower = np.array(lower, dtype=np.int64)
+
+    # need to convert action to fortran-style order and contiguous layout.
+    action_t = np.asfortranarray(np.transpose(action))
```

Comment:
Is this necessary?  It's difficult to dig through the code, but I think the `action` matrix is allocated in column-major order in the top-level calling functions, like `core.fitting.bspline_profile`.

Reply:
well, the original C code definitely did a transpose of the action matrix and `asfortranarray` is definitely required for tests to pass. in `bspline_profile`, order is set to `'F'` which sets up the memory format to be fortran-contiguous, but doesn't change the indexing. `transpose` also creates a new array and doesn't provide the `order` option. so that's why `asfortranarray` is required. i'll have a deeper look/think and see if there's some way to do this more cleanly.